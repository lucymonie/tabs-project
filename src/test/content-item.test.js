import React from 'react'
import { shallow } from 'enzyme'
import ContentItem from '../content-item'

const props = {
  article: {
    webUrl: 'something',
    webTitle: 'a lovely title',
    fields: {
      trailText: 'something else, something really overly long to fit in the space allowed in the trail text space, very long and totally over the top',
      wordcount: 3000
    }
  }
}

const component = shallow(<ContentItem {...props} />)

it('renders content items', () => {
  expect(component).toMatchSnapshot()
})

it('renders a title', () => {
  const title = component.find('a').at(0)
  expect(title.text()).toBe('a lovely title')
})

it('renders trauncated trail text', () => {
  const trail = component.find('span').at(0)
  expect(trail.text()).toBe('something else, something really overly long to fit in the space allowed...')
  const trailWords = trail.text().split(' ')
  expect(trailWords.length).toBeLessThan(13)
})

it('can calculate and render reading time', () => {
  const wordcount = component.find('span').at(1)
  expect(wordcount.text()).toBe('- 11 minute read')
})
