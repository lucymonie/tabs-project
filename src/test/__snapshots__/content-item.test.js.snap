// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders content items 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ContentItem
    article={
      Object {
        "fields": Object {
          "trailText": "something else, something really overly long to fit in the space allowed in the trail text space, very long and totally over the top",
          "wordcount": 3000,
        },
        "webTitle": "a lovely title",
        "webUrl": "something",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <a
          href="something"
          rel="noopener noreferrer"
          target="_blank"
        >
          a lovely title
        </a>,
        <br />,
        <span
          className="trailText"
        >
          something else, something really overly long to fit in the space allowed...
        </span>,
        <span
          className="readTime"
        >
          - 11 minute read
        </span>,
      ],
      "className": "not-last-list-item",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "a lovely title",
          "href": "something",
          "rel": "noopener noreferrer",
          "target": "_blank",
        },
        "ref": null,
        "rendered": "a lovely title",
        "type": "a",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "something else, something really overly long to fit in the space allowed...",
          "className": "trailText",
        },
        "ref": null,
        "rendered": "something else, something really overly long to fit in the space allowed...",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "- 11 minute read",
          "className": "readTime",
        },
        "ref": null,
        "rendered": "- 11 minute read",
        "type": "span",
      },
    ],
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <a
            href="something"
            rel="noopener noreferrer"
            target="_blank"
          >
            a lovely title
          </a>,
          <br />,
          <span
            className="trailText"
          >
            something else, something really overly long to fit in the space allowed...
          </span>,
          <span
            className="readTime"
          >
            - 11 minute read
          </span>,
        ],
        "className": "not-last-list-item",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "a lovely title",
            "href": "something",
            "rel": "noopener noreferrer",
            "target": "_blank",
          },
          "ref": null,
          "rendered": "a lovely title",
          "type": "a",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "something else, something really overly long to fit in the space allowed...",
            "className": "trailText",
          },
          "ref": null,
          "rendered": "something else, something really overly long to fit in the space allowed...",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "- 11 minute read",
            "className": "readTime",
          },
          "ref": null,
          "rendered": "- 11 minute read",
          "type": "span",
        },
      ],
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
